lib/libyui_client/wait.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module LibyuiClient
+  module Wait
+    def self.until(timeout:, interval:, message: nil, object: nil)
+      Timer.wait(timeout, interval) do
```

Comment:
sure, let's not go ahead so fast, that I have to "taste my own medicine" :)
I was thinking in this, but I haven't tested (where object is return in both cases):
```
def self.wait(timeout:, interval:, exit_condition:, message: nil, object: nil)
      Timer.wait(timeout, interval) do
        result = yield(object)
        return result if result == exit_condition
      end
      raise Error::TimeoutError, "timed out after #{timeout} seconds. #{message}"
    end

    def self.until(timeout:, interval:, message, object: nil)
      self.wait(timeout, interval, message, exit_condition: true, object)
    end

    def self.while(timeout:, interval:, message, object: nil)
      self.wait(timeout, interval, message, exit_condition: false, object)
    end
```

Reply:
It could work, but what are the benefits we'll get with this? First: 3 methods instead of two. Also, in case we'll apply this we have to make `def self.wait` to be `private`, as it should not be a part of the public API.

In my opinion it is just a matter of taste. We can apply this or we can leave it as is, not too much will be changed. If it will make your day happier, I can apply the solution:) 

So, WDYT?

Reply:
let's go for two methods then, looks better your original solution, (unlikely that something change there and refactor could have some potential value). thanks for considering.